cmake/EkatCreateUnitTest.cmake

```diff
@@ -68,73 +69,78 @@ function(EkatCreateUnitTest target_name target_srcs)
   string(STRIP "${ecut_LIBS_DIRS}" ecut_LIBS_DIRS)
   string(STRIP "${ecut_INCLUDE_DIRS}" ecut_INCLUDE_DIRS)
 
-  #---------------------------#
-  #   Create the executable   #
-  #---------------------------#
-
-  # Set link directories (must be done BEFORE add_executable is called)
-  # NOTE: CMake 3.15 adds 'target_link_directories', which is superior (does not pollute other targets).
-  #       We should switch to that as soon as we can assume CMAKE_VERSION >= 3.15.
-  if (ecut_LIBS_DIRS)
-    link_directories("${ecut_LIBS_DIRS}")
-  endif()
-  add_executable (${target_name} ${target_srcs})
+  if (NOT ecut_TEST_EXE)
```

Comment:
The code inside the `if` could be extracted into a separate macro (something like `EkatCreateUnitTestExec`, which EkatCreateUnitTest can call) so that in the app cmake we can do
```
# Create executable
EkatCreateUnitTestExec (my_exec my_exec.cpp LIBS "scream_share")

# Create tests
EkatCreateUnitTest(my_test_1 TEST_EXEC my_exec EXE_ARGS input_file1)
...
EkatCreateUnitTest(my_test_2 TEST_EXEC my_exec EXE_ARGS input_fileN)
```

Reply:
@bartgol , the usage based on the current state of the PR would be:

```
# Create tests
EkatCreateUnitTest(my_test_1 my_exec.cpp scream_share EXE_ARGS input_file1)
...
EkatCreateUnitTest(my_test_2 "" "" TEST_EXEC my_exec_1 EXE_ARGS input_fileN)
```

Dunno if this is better or not. It is a bit shorter and keeps the cmake unit test API at one function, so I'm leaning towards this approach.